src/main/java/com/google/api/codegen/configgen/InterfaceTransformer.java

```diff
@@ -24,4 +26,6 @@
    * unique resource paths to a short name used by the collection configuration.
    */
   Map<String, String> getResourceToEntityNameMap(InterfaceModel apiInterface);
+
+  void generateResourceNameGenerations(ConfigNode parentNode, ApiModel model);
```

Comment:
What does this mean? ("generate...Generations")

Reply:
`resource_name_generation` is a repeated field of the gapic config.